File: ts-sdk/src/sdk/models/operations/deletecollaborationsid.ts
import { SpeakeasyMetadata, SpeakeasyBase, RetryConfig } from "../../../internal/utils";
import * as shared from "../shared";



export class DeleteCollaborationsIdPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=collaboration_id" })
  collaborationId: string;
}


export class DeleteCollaborationsIdRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  retries?: RetryConfig;

  @SpeakeasyMetadata()
  pathParams: DeleteCollaborationsIdPathParams;
}


export class DeleteCollaborationsIdResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  clientError?: shared.ClientError;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;
}

File: go-sdk/pkg/models/operations/deletecollaborationsid.go
package operations

import (
	"openapi/pkg/models/shared"
	"openapi/pkg/utils"
)

type DeleteCollaborationsIDPathParams struct {
	CollaborationID string `pathParam:"style=simple,explode=false,name=collaboration_id"`
}

type DeleteCollaborationsIDRequest struct {
	Retries    *utils.RetryConfig
	PathParams DeleteCollaborationsIDPathParams
}

type DeleteCollaborationsIDResponse struct {
	ClientError *shared.ClientError
	ContentType string
	StatusCode  int64
}
